packages/zapp/console/src/components/Executions/NodeExecutionCacheStatus.tsx

```diff
@@ -68,20 +74,78 @@ export interface NodeExecutionCacheStatusProps {
  * the cache status with a descriptive message. For `Core.CacheCatalogStatus.CACHE_HIT`,
  * it will also attempt to render a link to the source `WorkflowExecution` (normal
  * variant only).
+ *
+ * For Map Tasks, we will check the NodeExecutionDetail for the cache status instead. Since map tasks
+ * cotains multiple tasks, the logic of the cache status is different.
  */
 export const NodeExecutionCacheStatus: React.FC<NodeExecutionCacheStatusProps> = ({
-  taskNodeMetadata,
+  execution,
   variant = 'normal',
+}) => {
+  const taskNodeMetadata = execution.closure?.taskNodeMetadata;
+  const detailsContext = useNodeExecutionContext();
+  const [nodeDetails, setNodeDetails] = React.useState<NodeExecutionDetails | undefined>();
+
+  React.useEffect(() => {
+    let isCurrent = true;
+    detailsContext.getNodeExecutionDetails(execution).then((res) => {
+      if (isCurrent) {
+        setNodeDetails(res);
+      }
+    });
+    return () => {
+      isCurrent = false;
+    };
+  });
+
+  console.log('testiung', nodeDetails);
+  if (nodeDetails?.taskTemplate?.type === TaskType.ARRAY) {
+    if (nodeDetails?.taskTemplate?.metadata?.cacheSerializable) {
+      return <CacheStatus cacheStatus={CatalogCacheStatus.MAP_CACHE} variant={variant} />;
+    }
+  }
+
+  if (taskNodeMetadata == null || taskNodeMetadata.cacheStatus == null) {
```

Comment:
nit: can be simplified to 
` if (taskNodeMetadata?.cacheStatus === null) {return null}`
or even ` if (!taskNodeMetadata?.cacheStatus) {return null}`

Reply:
fixed!